src/lib/pickles/wrap.mli

```diff
@@ -1,162 +1,132 @@
 open Pickles_types
+open Import
+
+type scalar_challenge_constant = Challenge.Constant.t Scalar_challenge.t
 
 val wrap :
-     max_proofs_verified:'max_proofs_verified Pickles_types.Nat.t
+     max_proofs_verified:'max_proofs_verified Nat.t
   -> (module Pickles_types.Hlist.Maxes.S
         with type length = 'max_proofs_verified
          and type ns = 'max_local_max_proofs_verifieds )
   -> ('max_proofs_verified, 'max_local_max_proofs_verifieds) Requests.Wrap.t
   -> dlog_plonk_index:
-       Backend.Tock.Curve.Affine.t Pickles_types.Plonk_verification_key_evals.t
-  -> (   ( Impls.Wrap.Impl.Field.t
-         , Impls.Wrap.Impl.Field.t Composition_types.Scalar_challenge.t
-         , Impls.Wrap.Impl.Field.t Pickles_types.Shifted_value.Type1.t
-         , ( Impls.Wrap.Impl.Field.t Pickles_types.Shifted_value.Type1.t
-           , Impls.Wrap.Impl.field Snarky_backendless.Cvar.t
-             Snarky_backendless.Snark_intf.Boolean0.t )
-           Pickles_types.Plonk_types.Opt.t
-         , ( Impls.Wrap.Impl.Field.t Composition_types.Scalar_challenge.t
-             Composition_types.Wrap.Proof_state.Deferred_values.Plonk.In_circuit
-             .Lookup
-             .t
-           , Impls.Wrap.Impl.field Snarky_backendless.Cvar.t
-             Snarky_backendless.Snark_intf.Boolean0.t )
-           Pickles_types.Plonk_types.Opt.t
+       Kimchi_pasta_basic.Pallas.Affine.t Plonk_verification_key_evals.t
```

Comment:
I would suppose we want to keep the curve name "abstract" with Tick/Tock.

Reply:
I propose to address this in https://github.com/MinaProtocol/mina/pull/14081